internal/asm/arm64/impl.go

```diff
@@ -0,0 +1,2340 @@
+			byte(skipOffset >> 16),
+			0x14,
+		})
+
+		// Then adding the consts into the binary.
+		for _, c := range a.pool.consts {
+			offsetOfConst := a.Buf.Len()
+			a.Buf.Write([]byte{byte(c), byte(c >> 8), byte(c >> 16), byte(c >> 24)})
+
+			// Invoke callbacks for `c` with the offset of binary where we store `c`.
+			for _, cb := range a.pool.offsetFinalizedCallbacks[c] {
+				cb(offsetOfConst)
+			}
+		}
+
+		// After the flush, reset the constant pool.
+		a.pool = constPool{offsetFinalizedCallbacks: map[int32][]func(int){}}
+	}
+}
+
+func (a *AssemblerImpl) setConstPoolCallback(v int32, cb func(int)) {
+	a.pool.offsetFinalizedCallbacks[v] = append(a.pool.offsetFinalizedCallbacks[v], cb)
+}
+
+func (a *AssemblerImpl) addConstPool(v int32, useOffset asm.NodeOffsetInBinary) {
+	if a.pool.firstUseOffsetInBinary == nil {
+		a.pool.firstUseOffsetInBinary = &useOffset
+	}
+
+	if _, ok := a.pool.offsetFinalizedCallbacks[v]; !ok {
+		a.pool.consts = append(a.pool.consts, v)
+		a.pool.offsetFinalizedCallbacks[v] = []func(int){}
+	}
+}
+
+// Bytes returns the encoded binary.
+func (a *AssemblerImpl) Bytes() []byte {
+	// 16 bytes alignment to match our impl with golang-asm.
+	// https://github.com/golang/go/blob/release-branch.go1.15/src/cmd/internal/obj/arm64/asm7.go#L62
+	//
+	// TODO: Delete after golang-asm removal.
+	if pad := 16 - a.Buf.Len()%16; pad > 0 && pad != 16 {
+		a.Buf.Write(make([]byte, pad))
+	}
+	return a.Buf.Bytes()
+}
+
+// EncodeNode encodes the given node into writer.
+func (a *AssemblerImpl) EncodeNode(n *NodeImpl) (err error) {
+	switch n.Types {
+	case OperandTypesNoneToNone:
+		err = a.EncodeNoneToNone(n)
+	case OperandTypesNoneToRegister, OperandTypesNoneToMemory:
+		err = a.EncodeJumpToRegister(n)
+	case OperandTypesNoneToBranch:
+		err = a.EncodeRelativeBranch(n)
+	case OperandTypesRegisterToRegister:
+		err = a.EncodeRegisterToRegister(n)
+	case OperandTypesLeftShiftedRegisterToRegister:
+		err = a.EncodeLeftShiftedRegisterToRegister(n)
+	case OperandTypesTwoRegistersToRegister:
+		err = a.EncodeTwoRegistersToRegister(n)
+	case OperandTypesThreeRegistersToRegister:
+		err = a.EncodeThreeRegistersToRegister(n)
+	case OperandTypesTwoRegistersToNone:
+		err = a.EncodeTwoRegistersToNone(n)
+	case OperandTypesRegisterAndConstToNone:
+		err = a.EncodeRegisterAndConstToNone(n)
+	case OperandTypesRegisterToMemory:
+		err = a.EncodeRegisterToMemory(n)
+	case OperandTypesMemoryToRegister:
+		err = a.EncodeMemoryToRegister(n)
+	case OperandTypesConstToRegister:
+		err = a.EncodeConstToRegister(n)
+	case OperandTypesSIMDByteToSIMDByte:
+		err = a.EncodeSIMDByteToSIMDByte(n)
+	case OperandTypesSIMDByteToRegister:
+		err = a.EncodeSIMDByteToRegister(n)
+	case OperandTypesTwoSIMDBytesToSIMDByteRegister:
+		err = a.EncodeTwoSIMDBytesToSIMDByteRegister(n)
+	default:
+		err = fmt.Errorf("encoder undefined for [%s] operand type", n.Types)
+	}
+	return
+}
+
+// CompileStandAlone implements asm.AssemblerBase.CompileStandAlone
```

Comment:
nested linking doesn't seem to navigate, which is why I have been switching to the following when implementing something out of package.

```
Xxx implements the same method as documented on yyy.Zzzz.
```